packages/fs-detectors/src/detect-framework.ts

```diff
@@ -135,6 +135,31 @@ async function matches(
   };
 }
 
+function removeSupercededFramework(
+  matches: (Pick<Framework, 'supercedes' | 'slug'> | null)[],
+  slug: string
+) {
+  const index = matches.findIndex(f => f?.slug === slug);
+  if (index !== -1) {
+    const framework = matches[index]!;
+    if (framework.supercedes) {
+      removeSupercededFramework(matches, framework.supercedes);
+    }
+    matches.splice(index, 1);
+  }
+}
+
+function removeSupercededFrameworks(
+  matches: (Pick<Framework, 'supercedes' | 'slug'> | null)[]
+) {
+  for (const match of matches.slice()) {
```

Comment:
Can you explain the `matches.slice()`. It seems like its meant to clone the array but the clone isn't used at all.

Reply:
Oh I think I'm confused because I thought `removeSupercededFrameworks()` was calling itself but the function call below is missing the `s` and therefore a different function <picture data-single-emoji=":doh:" title=":doh:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/doh/fd3ac3fe7525d318.jpg" alt=":doh:" align="absmiddle"></picture>